fdbbackup/backup.actor.cpp

```diff
@@ -3830,43 +3860,29 @@ int main(int argc, char* argv[]) {
 		};
 
 		auto initCluster = [&](bool quiet = false) {
-			auto resolvedClusterFile = ClusterConnectionFile::lookupClusterFileName(clusterFile);
-			try {
-				ccf = makeReference<ClusterConnectionFile>(resolvedClusterFile.first);
-			} catch (Error& e) {
-				if (!quiet)
-					fprintf(stderr, "%s\n", ClusterConnectionFile::getErrorString(resolvedClusterFile, e).c_str());
-				return false;
-			}
-
-			try {
-				db = Database::createDatabase(ccf, -1, IsInternal::True, localities);
-			} catch (Error& e) {
-				fprintf(stderr, "ERROR: %s\n", e.what());
-				fprintf(stderr, "ERROR: Unable to connect to cluster from `%s'\n", ccf->getFilename().c_str());
-				return false;
+			Optional<Database> result = connectToCluster(clusterFile, localities, quiet);
+			if (result.present()) {
+				db = result.get();
 			}
 
-			return true;
+			return result.present();
 		};
 
-		if (sourceClusterFile.size()) {
-			auto resolvedSourceClusterFile = ClusterConnectionFile::lookupClusterFileName(sourceClusterFile);
-			try {
-				sourceCcf = makeReference<ClusterConnectionFile>(resolvedSourceClusterFile.first);
-			} catch (Error& e) {
-				fprintf(stderr, "%s\n", ClusterConnectionFile::getErrorString(resolvedSourceClusterFile, e).c_str());
-				return FDB_EXIT_ERROR;
+		auto initSourceCluster = [&](bool required, bool quiet = false) {
+			if (!sourceClusterFile.size() && required) {
+				if (!quiet) {
+					fprintf(stderr, "ERROR: source cluster file is required\n");
```

Comment:
I sort of think that the destination cluster file should be required too, but that is not actually the case now and it will use the cluster file search path if you don't specify one. I decided not to change that here since that could in theory be a negative behavior change for someone.